frontend/src/scenes/dashboard/Dashboard.tsx

```diff
@@ -51,10 +51,14 @@ function DashboardScene(): JSX.Element {
         dashboardMode,
         receivedErrorsFromAPI,
     } = useValues(dashboardLogic)
-    const { setDashboardMode, setDates, reportDashboardViewed, setProperties } = useActions(dashboardLogic)
+    const { setDashboardMode, setDates, reportDashboardViewed, setProperties, abortAnyRunningQuery } =
+        useActions(dashboardLogic)
 
     useEffect(() => {
         reportDashboardViewed()
+        return () => {
```

Comment:
Suggestion: create a `useQueryCancellation(fn)`

Reply:
The problem with wrapping (or at least the problem I've not figured out the solution to) is the insight and dashboard cancellation cases are still really different